keps/sig-multicluster/2149-clusterid/README.md

```diff
@@ -341,68 +419,134 @@ Contains a unique identifier for the containing cluster.
+**Renaming a cluster**: Since a `cluster.clusterset.k8s.io ClusterProperty` must
+be immutable for the duration of its *membership* in a given ClusterSet, the
+property contents can be "changed" by unregistering the cluster from the
+ClusterSet and reregistering it with the new name.
+
+**Reusing cluster names**: Since an `cluster.clusterset.k8s.io ClusterProperty`
+has no restrictions on whether or not a ClusterProperty can be repeatable, if a
+cluster unregisters from a ClusterSet it is permitted under this standard to
+rejoin later with the same `cluster.clusterset.k8s.io ClusterProperty` it had
+before. Similarly, a *different* cluster could join a ClusterSet with the same
+`cluster.clusterset.k8s.io ClusterProperty` that had been used by another
+cluster previously, as long as both do not have membership in the same
+ClusterSet at the same time. Finally, two or more clusters may have the same
+`cluster.clusterset.k8s.io ClusterProperty` concurrently (though they **should**
+not; see "Uniqueness" above) *as long as* they both do not have membership in
+the same ClusterSet.
 
 **Cluster identifier composed of two DNS labels**: If a clusterset includes
 clusters registered with multiple cluster registries, the value of
 `cluster.clusterset.k8s.io ClusterProperty` may not uniquely identify a cluster
-within the clusterset. In such case the value of `cluster.clusterset.k8s.io ClusterProperty`
-may be composed of two DNS labels separated with a dot. The additional DNS label
-is implementation-dependent and allows the implementation to uniquely identify
-the cluster registry with which a cluster is registered.
+within the clusterset. In such case the value of `cluster.clusterset.k8s.io
+ClusterProperty` may be composed of two DNS labels separated with a dot. The
+additional DNS label is implementation-dependent and allows the implementation
+to uniquely identify the cluster registry with which a cluster is registered.
 
 #### Property: `clusterset.k8s.io`
 
-Contains an identifier that relates the containing cluster to the ClusterSet in which it belongs.
+Contains an identifier that relates the containing cluster to the ClusterSet in
+which it belongs.
 
 
 ##### Lifespan
 
-*   The identifier **must** exist and be immutable for the duration of a cluster’s membership in a ClusterSet.
-*   The identifier **must not** exist when the cluster is not a member of a ClusterSet.
+*   The identifier **must** exist and be immutable for the duration of a
+    cluster’s membership in a ClusterSet.
+*   The identifier **must not** exist when the cluster is not a member of a
+    ClusterSet.
 
 
 ##### Contents
 
 *   The identifier **must** associate the cluster with a ClusterSet.
-*   The identifier **may** be either unique or shared by members of a ClusterSet.
+*   The identifier **may** be either unique or shared by members of a
```

Comment:
Is this more simply stated as "The identifier **may** be any value" ?  Do we include non-ASCII?  We don't have a lot of concrete use for this yet, so it's hard to know if we are cutting off interesting exploration

Reply:
I would also add * The identifier **should** be the same across members of a ClusterSet" ?

Reply:
This was introduced to highlight two patterns:

1. where the identifier is the same for every member and directly identifies the clusterset
2. where the identifier is actually a reference to a relationship - i.e. points to a cluster's "membership"

This is slightly different than just "any value" as we steer away from a mixed case. May instead of should/must because we have no control and as you say, no concrete use. But it seems like making excess room for exploration here may also be making room for pain

Reply:
Why would `clusterset.k8s.io` indicate the relationship?  I may have missed something?